url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "WEDNESDAY"
+
+	// $ANTLR start "THURSDAY"
+	public final void mTHURSDAY() throws RecognitionException {
+		try {
+			int _type = THURSDAY;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:332:11: ( 'thursday' ( 's' )? | 'thur' ( DOT )? | 'thu' ( DOT )? | 'thus' ( DOT )? | 'thurs' ( DOT )? )
+			int alt60=5;
+			int LA60_0 = input.LA(1);
+			if ( (LA60_0=='t') ) {
+				int LA60_1 = input.LA(2);
+				if ( (LA60_1=='h') ) {
+					int LA60_2 = input.LA(3);
+					if ( (LA60_2=='u') ) {
+						switch ( input.LA(4) ) {
+						case 'r':
+							{
+							int LA60_4 = input.LA(5);
+							if ( (LA60_4=='s') ) {
+								int LA60_7 = input.LA(6);
+								if ( (LA60_7=='d') ) {
+									alt60=1;
+								}
+
+								else {
+									alt60=5;
+								}
+
+							}
+
+							else {
+								alt60=2;
+							}
+
+							}
+							break;
+						case 's':
+							{
+							alt60=4;
+							}
+							break;
+						default:
+							alt60=3;
+						}
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 60, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 60, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 60, 0, input);
+				throw nvae;
+			}
+
+			switch (alt60) {
+				case 1 :
+					// HiveLexer.g:332:13: 'thursday' ( 's' )?
+					{
+					match("thursday"); 
+
+					// HiveLexer.g:332:25: ( 's' )?
+					int alt55=2;
+					int LA55_0 = input.LA(1);
+					if ( (LA55_0=='s') ) {
+						alt55=1;
+					}
+					switch (alt55) {
+						case 1 :
+							// HiveLexer.g:332:25: 's'
+							{
+							match('s'); 
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:332:33: 'thur' ( DOT )?
+					{
+					match("thur"); 
+
+					// HiveLexer.g:332:40: ( DOT )?
+					int alt56=2;
+					int LA56_0 = input.LA(1);
+					if ( (LA56_0=='.') ) {
+						alt56=1;
+					}
+					switch (alt56) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 8418:8444, 8448:8474, 8478:8504, 8508:8534, 8801:8827, 8831:8857.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.